src/components/basic/display/Trades/useTrades.ts

```diff
@@ -0,0 +1,172 @@
+import { useContext, useEffect, useRef, useState } from "react";
+import { useRecoilCallback, useRecoilState } from "recoil";
+
+import { StrategyState } from "types";
+
+import {
+  EventWithBlockInfo,
+  fetchTradesAndReverts,
+  FetchTradesAndRevertsResult,
+  matchTradesAndReverts,
+  Trade,
+} from "api/web3/trades";
+
+import { lastCheckedBlockSelector, tradesSelector } from "state/atoms";
+
+import { useNewBlockHeader } from "hooks/useNewBlockHeader";
+
+import { TRADES_BATCH_SIZE } from "utils/constants";
+import getLoggerOrCreate from "utils/logger";
+
+import {
+  ContractInteractionContext,
+  ContractInteractionContextProps,
+} from "components/context/ContractInteractionProvider";
+
+const logger = getLoggerOrCreate("use trades hook");
+
+export function useTrades(
+  strategy: StrategyState
+): { trades: Trade[]; isLoading: boolean } {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [trades, setTrades] = useRecoilState(
+    tradesSelector(strategy.transactionHash)
+  );
+
+  const context = useContext(ContractInteractionContext);
+
+  // "instance" variable to control when component is unmounted and drop promises
+  const cancelled = useRef(false);
+  // "instance" variable to prevent concurrent queries
+  const canQuery = useRef(true);
+
+  const newBlock = useNewBlockHeader();
+
+  const fetchAll = useRecoilCallback(
+    ({ snapshot, set }) => async (
+      strategy: StrategyState,
+      context: ContractInteractionContextProps,
+      _latestBlockNumber?: number
```

Comment:
I thought `_` "dangling underscore" was against linting rules. Do you want to keep it?